Kconfig.zephyr

```diff
@@ -19,7 +19,7 @@ osource "$(KCONFIG_BINARY_DIR)/Kconfig.shield.defconfig"
 # This loads Zephyr base shield defconfigs
 source "boards/shields/*/Kconfig.defconfig"
 
-source "$(BOARD_DIR)/Kconfig.defconfig"
+osource "$(BOARD_DIR)/Kconfig.defconfig"
```

Comment:
this effectively changes `Kconfig.defconfig` from being mandatory to being optional.

The alternative approach is to keep the `source` and then boards not having any default to override  must create an empty file, maybe containing `# Intentionally left empty`.

I have a slight preference for keeping the `source` and then an empty file for those boards not needing to override anything.

Reason being:
- most downstream users of Zephyr will at some point need to create a custom board of there own, and having a `source` helps them to place board specific defconfigs the right place.
- It prevents stupid user mistakes like writing `kconfig.defconfig` / `Kconfig.defconf`instead of `Kconfig.defconfig`.
- If board is later updated to override defaults then it's clearer where to do this if the file is already empty (but present)

Reply:
fair enough, we have different opinion on this.

Another risk is that someone later wants to add a new `default` to a board which doesn't have a `Kconfig.defconfig` already.
Contributor looks around in that board folder and decides to place it in `Kconfig.$(BOARD)` because lack of knowledge.
Such PR can easily approved and be merged, resulting in inconsistency.

Whereas a `Kconfig.defconfig` with content:
```
# Kconfig.defconfig file for board specific overload of default settings.

# No board specific defaults.
```

will help to ensure that such defaults goes to the right location.

Contributors will often look for an existing file to place small suggestions, rather than create new files.